data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1138 @@
+table.insert(chemical, hazardous_substances)
+
+local mining = {}
+table.insert(mining, clus)
+table.insert(mining, explosives)
+
+local hardware = {}
+table.insert(hardware, aluminium_tubes)
+table.insert(hardware, diamonds)
+table.insert(hardware, hazardous_substances)
+table.insert(hardware, machine_tools)
+table.insert(hardware, neptunium)
+table.insert(hardware, plutonium)
+table.insert(hardware, semi_finished_products)
+table.insert(hardware, spaceship_parts)
+table.insert(hardware, titanium)
+table.insert(hardware, tungsten)
+table.insert(hardware, uranium)
+
+local infrastructure = {}
+table.insert(infrastructure, clus)
+table.insert(infrastructure, explosives)
+table.insert(infrastructure, greenhouses)
+
+local consumer_goods = {}
+table.insert(consumer_goods, electrical_appliances)
+table.insert(consumer_goods, furniture)
+table.insert(consumer_goods, spaceship_parts)
+
+local expensive = {} -- price >= 175
+table.insert(expensive, art_objects)
+table.insert(expensive, diamonds)
+table.insert(expensive, neptunium)
+table.insert(expensive, plutonium)
+table.insert(expensive, spaceship_parts)
+table.insert(expensive, uranium)
+
+local fluffy = {}
+table.insert(fluffy, quibbles)
+
+local wedding = {}
+table.insert(wedding, wedding_dresses)
+
+local art = {}
+table.insert(art, art_objects)
+
+local gems = {}
+table.insert(gems, diamonds)
+
+local radioactive = {}
+table.insert(radioactive, neptunium)
+table.insert(radioactive, plutonium)
+table.insert(radioactive, uranium)
+
+local centrifuges = {}
+table.insert(centrifuges, aluminium_tubes)
+table.insert(centrifuges, machine_tools)
+table.insert(centrifuges, titanium)
+
+local custom_cargo = {
+	{ bkey = "CHEMICAL", goods = chemical, weight },
+	{ bkey = "MINING", goods = mining, weight },
+	{ bkey = "HARDWARE", goods = hardware, weight },
+	{ bkey = "INFRASTRUCTURE", goods = infrastructure, weight },
+	{ bkey = "CONSUMER_GOODS", goods = consumer_goods, weight },
+	{ bkey = "EXPENSIVE", goods = expensive, weight },
+	{ bkey = "FLUFFY", goods = fluffy, weight },
+	{ bkey = "WEDDING" , goods = wedding, weight },
+	{ bkey = "ART", goods = art, weight },
+	{ bkey = "GEMS", goods = gems, weight },
+	{ bkey = "RADIOACTIVE", goods = radioactive, weight },
+	{ bkey = "CENTRIFUGES", goods = centrifuges, weight }
+}
+
+-- Each branch should have a probability weight proportional to its size
+local custom_cargo_weight_sum = 0
+for branch,branch_array in pairs(custom_cargo) do
+	custom_cargo[branch].weight = #branch_array.goods
+	custom_cargo_weight_sum = custom_cargo_weight_sum + #branch_array.goods
+end
+
+local ads = {}
+local missions = {}
+
+local isQualifiedFor = function(reputation, ad)
+	return
+		reputation >= 12 or
+		ad.localdelivery or
+		(ad.risk <  0.1 and ad.urgency <= 0.1) or
+		(ad.risk <  0.5 and ad.urgency <= 0.5 and reputation >= 4) or
+		false
+end
+
+local getNumberOf = function (str)
+	local num = 0
+
+	while l[str .. "_" .. num + 1] do
+		num = num + 1
+	end
+	return num
```

Comment:
There is a number of things that are wrong with this function.

First, instead of starting at 0 and always use the +1 value, you could start at 1, and just decrement at the end.

Second: What's that business with the translation table ? I am supposed to understand what is the purpose of the function when reading it, and yet I have no idea what is the intended use so far. My educated guess would be that you are trying to figure out how many flavours there are of some string. If that is the case, well... I find that whole business terribly hackish, but I don't think we have any other way to figure it out with the current state of affairs.

However, I'd like us to introduce an API within our translation module to do just that, as it seems to me that this thing might be a recurrent problem. That shouldn't be very hard to do, but that is not the scope of this PR.

In the mean time, would you be so kind as to document this function and/or rename it to a more explicit moniker?